lib/lp_token_auth/config.rb

```diff
@@ -9,22 +9,29 @@ class Config
     # * `expires` is an integer corresponding to the number of hours that the token is active
     # * `secret` is a string corresponding to the secret key used when encrypting the token
     # * `token_transport` is a string indicating where to include the token in the HTTP response
-    attr_accessor :algorithm, :expires, :secret, :token_transport
+    attr_accessor :algorithm, :expires, :secret, :token_transport, :jwe_private_key, :jwe_encryption
 
     # Provides default values to token options
+    # ENV defaults defined as procs to ensure they return their
+    # latest value at call time (else they return nil,
+    # since ENV values may not be initialized before gem code)
     DEFAULT_VALUES = {
       algorithm: 'HS512',
       expires: (7 * 24),
       token_transport: [:cookie],
+      jwe_private_key: -> { ENV['JWE_PRIVATE_KEY'] },
```

Comment:
do these need to be procs to evaluate the expression?

Reply:
explained in the comment above, added as procs because otherwise the constant is loaded before the rails environment has initialized the ENV var values, making the defaults nil. with proc, it'll call the current value when accessing `DEFAULT_VALUES` in any of the methods. 

also considered adding it as a class method, but then the hash would get recreated every time for the other default values not coming from ENV. which is probably not a huge deal, but this minimizes that while keeping all the possible defaults in one easy to find place (instead of within specific methods like before)